tracing/otel/helpers.go

```diff
@@ -21,3 +25,36 @@ func SpanRecordError(span trace.Span, err error, description string, eventOption
 	)
 	span.RecordError(err, eventOptions...)
 }
+
+var defaultGCPTraceLog = gcpTraceLog{
+	spanFieldName:  "logging.googleapis.com/spanId",
+	traceFieldName: "logging.googleapis.com/trace",
+}
+
+type gcpTraceLog struct {
+	traceFieldName string
+	spanFieldName  string
+	gcpProjectID   string
+}
+
+// LogFromCtx returns a log from the provided context. It adds GCP trace and span fields so the log can be associated with cloud tracing
+func (tl *gcpTraceLog) LogFromCtx(ctx context.Context) *zerolog.Logger {
+	log := zerolog.Ctx(ctx)
+	span := trace.SpanFromContext(ctx)
+	fields := map[string]interface{}{}
+	if span.SpanContext().HasSpanID() {
+		fields[tl.spanFieldName] = span.SpanContext().SpanID()
+	}
+	if span.SpanContext().HasTraceID() {
```

Comment:
Potentially should add `&& tl.gcpProjectID != ""`

Reply:
Yeh fair enough, I'll add that